VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java

```diff
@@ -289,8 +289,12 @@ public Fragment getItem(int pos) {
             CourseComponent unit = getUnit(pos);
             CourseUnitFragment unitFragment;
             //FIXME - for the video, let's ignore studentViewMultiDevice for now
-            if (unit instanceof VideoBlockModel && ((VideoBlockModel)unit).getData().encodedVideos.getPreferredVideoInfo() != null) {
-                unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit);
+            if (unit instanceof VideoBlockModel &&
+                    ((VideoBlockModel) unit).getData().encodedVideos.getPreferredVideoInfo() != null) {
+                    unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit);
+            } else if (unit instanceof VideoBlockModel &&
```

Comment:
The safer course seems to be the current implementation, which does require this check being performed potentially twice for simplicity of the execution path.